libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -606,17 +606,33 @@ bool IRGeneratorForStatements::visit(BinaryOperation const& _binOp)
 			solAssert(false, "Unknown comparison operator.");
 		define(_binOp) << expr << "\n";
 	}
-	else if (TokenTraits::isShiftOp(op) || op == Token::Exp)
+	else if (op == Token::Exp)
 	{
 		IRVariable left = convert(_binOp.leftExpression(), *commonType);
 		IRVariable right = convert(_binOp.rightExpression(), *type(_binOp.rightExpression()).mobileType());
-		if (op == Token::Exp)
+
+		if (auto literal = dynamic_cast<Literal const*>(&_binOp.leftExpression()))
```

Comment:
Wouldn't it be better to check if the type of leftExpression is RationalNumberType?